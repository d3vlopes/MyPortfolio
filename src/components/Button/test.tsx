import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
