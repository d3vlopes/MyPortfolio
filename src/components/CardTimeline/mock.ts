import { CardTimelineProps } from '.'

export default [
  {
    title: 'O início',
    content: `<p>Desde de muito cedo sempre gostei muito de tecnologia,
    como video game, computador, celular,
    mais nunca tinha tido interesse de querer saber como tudo aquilo funcionava.</p>
    <p>Meu primeiro contato com programação,
    foi durante o curso onde me formei Técnico em Informática pelo Senac.</p>
    <p>Foi algo mágico, eu digo que foi “amor a primeira linha de código”,
    eu me apaixonei por como tudo aquilo funcionava,  como cada tag fazia algo diferente,
    como a gente podia transformar o visual de um documento HTML, com a adição do CSS.
    Eu fiquei maravilhado todo aquele mundo e
    a infinidade de possibilidades que eu tinha descoberto.</p>
    `,
  },
  {
    title: 'InFome',
    content: `<p>Durante o curso, participei de alguns projetos com meus colegas,
    o principal deles foi a criação do InFome,
    um projeto com a ideia de divulgar lugares pouco conhecidos da cidade,
    mostrando informações, imagens, localizção,
    incentivando o usuário ir até o lugar divulgado.</p>
    <p>O projeto chegou a ser pré-incubado pela ITUNISC,
    que incentiva micro e pequenas empresas tecnologicamente inovadoras.</p>
    <p>Fui responsável por todo o desenvolvimento da aplicação,
    backend e frontend, junto com a cordenação do meu professor.</p>
    <p>Infelizmente o projeto não foi em frente,
    mais todo o aprendizado e vivência que tive, foi algo que me agregou muito.</p>
    `,
  },
  {
    title: 'Primeiro emprego',
    content: `<p>Após a formação, tive a oportunidade de trabalhar em uma
    agência de desenvolvimento aqui da cidade a Empório Adamantis.
    Foi como um sonho realizado estar ali dentro e poder ver de perto todo o processo
    de criação, desde do briefing, design até o desenvolvimento.</p>
    <p>Trabalhei lá como desenvolvedor Full Stack, atuava tanto no backend,
    como no frontend, criando novas aplicações ou fazendo alterações em alguma
    que já estava em produção quando necessário.</p>
    <p>Fiquei lá um período curto, mais foi uma experiência muito importante para mim,
    me ajudou a se transformar no profissional que sou hoje.</p>
    `,
  },
  {
    title: 'Frontend e Design',
    content: `<p>Depois da saida da Empório, foquei em estudar bastante,
    aprender novas tecnologias e me preparar bem para voltar ao mercado de trabalho.</p>
    <p>Foi ai que decidi me especializar no frontend,
    que era a parte do desenolvimento que eu mais gostava de trabalhar.</p>
    <p>E essa escolha de se especializar apenas em uma área do desenvolvimento
    também me ajudou a poder fazer outra coisa que eu amava,
    que era trabalhar com design e criação de interfaces.</p>
    <p>Eu sempre gostei muito de criar coisas,
    desenhar e era algo que eu tinha muita vontade de estudar e me especializar.</p>
    <p>Como os profissionais de UI e Frontend trabalham próxmos,
    foi perfeito conseguir juntar duas coisas que eu amo e trabalhar com elas juntas.</p>
    `,
  },
] as CardTimelineProps[]
