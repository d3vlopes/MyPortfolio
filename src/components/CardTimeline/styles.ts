import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    background: ${theme.colors.bg.secondary.dark};
    border-radius: ${theme.border.radius.medium};
    padding: ${theme.spacings.small} ${theme.spacings.large}
      ${theme.spacings.small};

    width: 100%;
    max-width: 86.7rem;

    &::before {
      position: absolute;
      content: '';
      border-bottom: 1px solid ${theme.colors.bg.secondary.light};
      top: 10.5rem;
      left: 0;
      right: 0;
    }
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-family: ${theme.font.heading};
    font-size: ${theme.font.sizes.xxlarge};
    color: ${theme.colors.text.light};
    padding: ${theme.spacings.small} 0 ${theme.spacings.small}
      ${theme.spacings.large};
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.large} ${theme.spacings.large}
      calc(${theme.spacings.large} - 1.6rem);
  `}
`
