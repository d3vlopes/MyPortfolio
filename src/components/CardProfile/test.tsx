import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
