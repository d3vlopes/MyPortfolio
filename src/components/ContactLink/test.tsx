import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Props } from 'react-inlinesvg'

import ContactLink, { ContactLinkProps } from '.'

const { getByLabelText } = screen

const mock: ContactLinkProps = {
  icon: '/img/icon.svg',
  name: 'Github',
  link: 'www.github.com/user/name',
}

jest.mock('react-inlinesvg', () => {
  function ReactInlineSVG({ name }: Props) {
    return <svg aria-label={name} />
  }

  return ReactInlineSVG
})

describe('<ContactLinks />', () => {
  it('should render ContactLinks', () => {
    const { container } = renderWithTheme(<ContactLink {...mock} />)

    const icon = getByLabelText(mock.name)
    const a = container.querySelector('a')

    expect(icon).toBeInTheDocument()
    expect(a).toHaveAttribute('href', mock.link)
  })
})
