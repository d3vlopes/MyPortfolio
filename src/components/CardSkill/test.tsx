import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
