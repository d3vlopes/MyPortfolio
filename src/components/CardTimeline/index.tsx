import TextContent from 'components/TextContent'

import * as S from './styles'

export type CardTimelineProps = {
  title: string
  content: string
}

const CardTimeline = ({ title, content }: CardTimelineProps) => {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <TextContent content={content} />
    </S.Wrapper>
  )
}

export default CardTimeline
