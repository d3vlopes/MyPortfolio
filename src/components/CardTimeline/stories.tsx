import { Story, Meta } from '@storybook/react'

import CardTimeline, { CardTimelineProps } from '.'
import mock from './mock'

export default {
  title: 'Components/CardTimeline',
  component: CardTimeline,
  parameters: {
    layout: 'centered',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=1043%3A1149',
    },
  },
  args: mock[0],
} as Meta<CardTimelineProps>

export const Default: Story<CardTimelineProps> = (args) => {
  return (
    <div>
      <CardTimeline {...args} />
    </div>
  )
}
