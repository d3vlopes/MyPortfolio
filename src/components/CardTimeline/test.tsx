import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardTimeline, { CardTimelineProps } from '.'

const props: CardTimelineProps = {
  title: 'CardTimeline',
  content: '<p>Loren ipsum dolor</p>',
}

const { getByRole, getByText } = screen

describe('<CardTimeline />', () => {
  it('should render CardTimeline', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    const title = getByRole('heading', { name: 'CardTimeline' })
    const content = getByText('Loren ipsum dolor')
    const p = container.querySelector('p')

    expect(title).toBeInTheDocument()
    expect(content).toBeInTheDocument()
    expect(p?.tagName.toLowerCase()).toBe('p')
  })

  it('should match snapshot ', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
