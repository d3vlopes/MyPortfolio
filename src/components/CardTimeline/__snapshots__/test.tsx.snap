// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTimeline /> should match snapshot  1`] = `
.c2 {
  font-size: 2.2rem;
  line-height: 180%;
  color: #DEE7EA;
}

.c2 h1,
.c2 h2,
.c2 h3,
.c2 h4,
.c2 h5,
.c2 h6 {
  font-family: Gilroy-Bold;
  color: #FFFFFF;
  margin: 1.6rem 0;
}

.c2 p {
  margin-bottom: 1.6rem;
  max-width: 59.7rem;
}

.c2 a,
.c2 a:visited,
.c2 a:link {
  color: #6D59A8;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c2 a:hover {
  color: #543F8C;
}

.c2 a:hover:hover > img {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c2 img {
  max-width: 100%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c2 img:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c2 .image {
  line-height: 0;
  margin: 2.4rem 0;
}

.c2 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c2 .image-half {
  max-width: 50%;
}

.c2 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c2 ul,
.c2 ol {
  padding: 0 2.4rem 1.6rem;
}

.c2 hr {
  border: none;
  border-bottom: 0.1rem solid #3B3A51;
}

.c2 code {
  font-family: monospace;
  color: #DEE7EA;
  font-size: 1.6rem;
  background: #302F43;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c2 pre {
  background: #302F43;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c2 pre code {
  color: inherit;
  background: inherit;
}

.c2 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c2 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c2 table td,
.c2 table th {
  padding: 1.6rem;
  border-bottom: 0.1rem solid #3B3A51;
}

.c2 blockquote {
  border-left: 0.5rem solid #6D59A8;
  color: #DEE7EA;
  padding-left: 2.4rem;
  margin: 2.4rem;
}

.c0 {
  position: relative;
  background: #302F43;
  border-radius: 0.6rem;
  padding: 2.4rem 4.0rem 2.4rem;
  width: 100%;
  max-width: 86.7rem;
}

.c0::before {
  position: absolute;
  content: '';
  border-bottom: 1px solid #484762;
  top: 10.5rem;
  left: 0;
  right: 0;
}

.c0 .content {
  padding: 4.0rem 0;
}

.c1 {
  font-family: Gilroy-Bold;
  font-size: 2.6rem;
  color: #FFFFFF;
  padding: 2.4rem 0;
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c2 img:hover {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@media (max-width:768px) {
  .c2 .image-style-side {
    max-width: 100%;
    float: none;
  }
}

@media (max-width:768px) {
  .c1 {
    font-size: 2rem;
  }
}

@media (max-width:450px) {
  .c1 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    CardTimeline
  </h2>
  <div
    class="c2"
  >
    <div
      class="content"
    >
      <p>
        Loren ipsum dolor
      </p>
    </div>
  </div>
</div>
`
